controllers/provisioners/eks/helpers.go

```diff
@@ -502,8 +502,15 @@ func (ctx *EksInstanceGroupContext) GetComputedBootstrapOptions() *v1alpha1.Boot
 		}
 
 		instanceTypeNetworkInfo := awsprovider.GetInstanceTypeNetworkInfo(state.GetInstanceTypeInfo(), configuration.InstanceType)
-		maxPods := (*instanceTypeNetworkInfo.MaximumNetworkInterfaces-1)*
-			(*instanceTypeNetworkInfo.Ipv4AddressesPerInterface-1) + hostNetworkPods
+		var prefixAssignmentEnabled = instanceGroup.GetAnnotations()[CustomNetworkingPrefixAssignmentEnabledAnnotation] == "true"
+		var maxPods int64 = 0
+		var enis = *instanceTypeNetworkInfo.MaximumNetworkInterfaces-1 //Primary interface is not used for pod networking when custom networking is enabled
+		if prefixAssignmentEnabled {
+			var ipsPerDelegatedPrefix int64 = 16 //Prefixes are /28 blocks, which contain 2^4 IPs
+			maxPods = (enis * ((*instanceTypeNetworkInfo.Ipv4AddressesPerInterface-1) * ipsPerDelegatedPrefix)) + hostNetworkPods
```

Comment:
Will be nice to have a variable declaration for ipsPerInterface above which can be referenced here and below instead of doing multiple ptr derefs here.